src/consensus/pbft/libbyz/new_view.cpp

```diff
@@ -106,5 +109,32 @@ bool New_view::pre_verify()
     return false;
   }
 
-  return true;
+  // check that New_view has 2f + 1 valid View_change digest signatures
+  auto proof_count = 0;
+  for (int i = 0; i < pbft::GlobalState::get_node().num_of_replicas(); i++)
+  {
+    if (vc_info()[i].d == Digest())
+    {
+      // we don't have a view change message from this replica
+      continue;
+    }
+    auto sender_principal = pbft::GlobalState::get_node().get_principal(i);
+    if (!sender_principal)
+    {
+      LOG_INFO_FMT("Sender principal has not been configured yet {}", i);
+      continue;
+    }
+    if (sender_principal->verify_signature(
+          vc_info()[i].d.digest(),
+          vc_info()[i].d.digest_size(),
+          vc_info()[i].sig.data(),
+          vc_info()[i].sig_size,
+          true /* allow self*/))
+    {
+      proof_count++;
+    }
+  }
+  LOG_TRACE_FMT("new view has verified {} view change messages", proof_count);
+
+  return proof_count >= (2 * pbft::GlobalState::get_node().f() + 1);
```

Comment:
node has a threshold variable that does the same thing

Reply:
thanks!